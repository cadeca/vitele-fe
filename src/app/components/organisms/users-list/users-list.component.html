<div class="users-list">
  <mat-list role="list">
    <mat-list-item role="listitem" class="list-headline">
      <div class="user-entry row font-weight-bold">
        <div class="col-3">
          {{ 'userDetail.firstName' | translate }}
        </div>
        <div class="col-2">
          {{ 'userDetail.lastName' | translate }}
        </div>
        <div class="col-3">
          {{ 'userDetail.username' | translate }}
        </div>
        <div class="col-4">
          {{ 'userDetail.email' | translate }}
        </div>
      </div>
    </mat-list-item>
    <div class="scrollable-list">
     <mat-list-item role="listitem" *ngFor="let user of visibleUsers">
      <div class="user-entry row">
        <div class="col-12 col-md-3">
          <span class="label">{{ 'userDetail.firstName' | translate }}:</span>
          {{ user.firstName }}
        </div>
        <div class="col-12 col-md-2">
          <span class="label">{{ 'userDetail.lastName' | translate }}:</span>
          {{ user.lastName }}
        </div>
        <div class="col-12 col-md-3">
          <span class="label">{{ 'userDetail.username' | translate }}:</span>
          {{ user.username }}
        </div>
        <div class="col-12 col-md-4">
          <span class="label">{{ 'userDetail.email' | translate }}:</span>
          {{ user.email }}
        </div>
      </div>
    </mat-list-item>
    </div>
    <mat-list-item role="listitem" *ngIf="!visibleUsers || !visibleUsers.length">
      {{ 'userDetail.noUser' | translate }}</mat-list-item>
  </mat-list>

</div>
