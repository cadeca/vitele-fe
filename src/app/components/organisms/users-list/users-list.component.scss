@import "src/assets/style/variables";

.user-entry {
  width: 100%;
  padding: 0.5rem 0;
}

/deep/ .mat-list-item {
  height: auto !important;
  margin-bottom: 15px;

  &:not(:first-child) {

    &:hover {
      box-shadow: $elevation-shadow-2px;
      cursor: pointer;
    }
  }

  .mat-list-item-content {
    height: auto !important;
  }
}

.user-entry .label {
  display: none;
}

.list-headline {
  box-shadow: $elevation-shadow-2px;
}

@media (max-width: $md) {

  .list-headline {
    display: none !important;
  }

  .user-entry .label {
    display: inline-flex;
  }

  .mat-list-item {
    box-shadow: $elevation-shadow-2px;

    &:hover {
      box-shadow: $elevation-shadow-8px;
    }
  }
}
