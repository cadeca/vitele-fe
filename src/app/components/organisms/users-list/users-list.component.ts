import {Component, Input} from '@angular/core';
import {User} from '../../../providers/types/wl-types';
import {Roles} from '../../../directives/roles';
import {Router} from '@angular/router';

@Component({
  selector: 'wl-users-list',
  templateUrl: './users-list.component.html',
  styleUrls: ['./users-list.component.scss']
})
export class UsersListComponent {
  visibleUsers: User[] = [];
  private allUsers: User[] = [];
  roles = Roles;


  @Input()
  set users(users: User[]) {
    if (users) {
      this.visibleUsers = [...users];
      this.allUsers = [...users];
    }
  }

  @Input()
  set searchField(searchField: string) {
    if (searchField) {
      this.visibleUsers = this.allUsers.filter(user => {
        return user.firstName.toLowerCase().includes(searchField.toLowerCase()) ||
          user.lastName.toLowerCase().includes(searchField.toLowerCase()) ||
          user.username.toLowerCase().includes(searchField.toLowerCase()) ||
          user.email.toLowerCase().includes(searchField.toLowerCase());
      });
    } else {
      this.visibleUsers = this.allUsers;
    }
  }

  constructor(private router: Router) { }

}
