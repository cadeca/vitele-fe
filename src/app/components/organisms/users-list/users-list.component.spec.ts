import {async, fakeAsync, TestBed, tick} from '@angular/core/testing';
import {MaterialModule} from '../../../material.module';
import {UsersListComponent} from './users-list.component';
import {TranslateModule} from '@ngx-translate/core';

describe('UsersListComponent', () => {
  let fixture;
  let component;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        MaterialModule,
        TranslateModule.forRoot()
      ],
      declarations: [UsersListComponent]
    })
      .compileComponents().then(() => {
      fixture = TestBed.createComponent(UsersListComponent);
      component = fixture.componentInstance;
      fixture.detectChanges();
    });
  }));

  it('should check initial state', () => {
    expect(component.isEditable).toBe(false);
    expect(component.visibleUsers.length).toBe(0);
  });

  it('should match empty snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should match snapshot with data', () => {
    component.visibleUsers = [
      {
        firstName: 'Test',
        lastName: 'User',
        username: 'test.user',
        email: 'test.user@email.com',
        type: 'student'
      },
      {
        firstName: 'Test1',
        lastName: 'User1',
        username: 'test1.user1',
        email: 'test1.user1@email.com',
        type: 'teacher'
      }
    ];
    expect(fixture).toMatchSnapshot();
  });
});
