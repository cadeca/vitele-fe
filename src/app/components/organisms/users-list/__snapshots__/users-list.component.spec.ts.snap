// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersListComponent should match empty snapshot 1`] = `
<wl-users-list
  allUsers={[Function Array]}
  isEditable="false"
  removeUser={[Function EventEmitter_]}
  visibleUsers={[Function Array]}
>
  <div
    class="users-list"
  >
    <mat-list
      class="mat-list mat-list-base"
      role="list"
    >
      <mat-list-item
        class="list-headline mat-list-item mat-focus-indicator"
        role="listitem"
      >
        <div
          class="mat-list-item-content"
        >
          <div
            class="mat-list-item-ripple mat-ripple"
            mat-ripple=""
          />
          <div
            class="mat-list-text"
          />
          <div
            class="user-entry row font-weight-bold "
          >
            <div
              class="col-2"
            >
               userDetail.firstName 
            </div>
            <div
              class="col-2"
            >
               userDetail.lastName 
            </div>
            <div
              class="col-3"
            >
               userDetail.username 
            </div>
            <div
              class="col-4"
            >
               userDetail.email 
            </div>
          </div>
        </div>
      </mat-list-item>
      <div
        class="scrollable-list"
      >
        
      </div>
      
      <mat-list-item
        class="mat-list-item mat-focus-indicator"
        role="listitem"
      >
        <div
          class="mat-list-item-content"
        >
          <div
            class="mat-list-item-ripple mat-ripple"
            mat-ripple=""
          />
          <div
            class="mat-list-text"
          />
           userDetail.noUser
        </div>
      </mat-list-item>
    </mat-list>
  </div>
</wl-users-list>
`;

exports[`UsersListComponent should match snapshot 1`] = `
<wl-users-list
  allUsers={[Function Array]}
  isEditable="false"
  removeUser={[Function EventEmitter_]}
  visibleUsers={[Function Array]}
>
  <div
    class="users-list"
  >
    <mat-list
      class="mat-list mat-list-base"
      role="list"
    >
      <mat-list-item
        class="list-headline mat-list-item mat-focus-indicator"
        role="listitem"
      >
        <div
          class="mat-list-item-content"
        >
          <div
            class="mat-list-item-ripple mat-ripple"
            mat-ripple=""
          />
          <div
            class="mat-list-text"
          />
          <div
            class="user-entry row font-weight-bold "
          >
            <div
              class="col-2"
            >
               userDetail.firstName 
            </div>
            <div
              class="col-2"
            >
               userDetail.lastName 
            </div>
            <div
              class="col-3"
            >
               userDetail.username 
            </div>
            <div
              class="col-4"
            >
               userDetail.email 
            </div>
          </div>
        </div>
      </mat-list-item>
      <div
        class="scrollable-list"
      >
        
      </div>
      
      <mat-list-item
        class="mat-list-item mat-focus-indicator"
        role="listitem"
      >
        <div
          class="mat-list-item-content"
        >
          <div
            class="mat-list-item-ripple mat-ripple"
            mat-ripple=""
          />
          <div
            class="mat-list-text"
          />
           userDetail.noUser
        </div>
      </mat-list-item>
    </mat-list>
  </div>
</wl-users-list>
`;

exports[`UsersListComponent should match snapshot with data 1`] = `
<wl-users-list
  allUsers={[Function Array]}
  isEditable="false"
  removeUser={[Function EventEmitter_]}
  visibleUsers={[Function Array]}
>
  <div
    class="users-list"
  >
    <mat-list
      class="mat-list mat-list-base"
      role="list"
    >
      <mat-list-item
        class="list-headline mat-list-item mat-focus-indicator"
        role="listitem"
      >
        <div
          class="mat-list-item-content"
        >
          <div
            class="mat-list-item-ripple mat-ripple"
            mat-ripple=""
          />
          <div
            class="mat-list-text"
          />
          <div
            class="user-entry row font-weight-bold "
          >
            <div
              class="col-2"
            >
               userDetail.firstName 
            </div>
            <div
              class="col-2"
            >
               userDetail.lastName 
            </div>
            <div
              class="col-3"
            >
               userDetail.username 
            </div>
            <div
              class="col-4"
            >
               userDetail.email 
            </div>
          </div>
        </div>
      </mat-list-item>
      <div
        class="scrollable-list"
      >
        
      </div>
      
      <mat-list-item
        class="mat-list-item mat-focus-indicator"
        role="listitem"
      >
        <div
          class="mat-list-item-content"
        >
          <div
            class="mat-list-item-ripple mat-ripple"
            mat-ripple=""
          />
          <div
            class="mat-list-text"
          />
           userDetail.noUser
        </div>
      </mat-list-item>
    </mat-list>
  </div>
</wl-users-list>
`;
